R/run-nmtran.R

```diff
@@ -0,0 +1,223 @@
+#' Interface for running `NM-TRAN` on model objects
+#'
+#' Provides functions to run `NM-TRAN` on model objects for validation and to
+#' generate `NM-TRAN` datasets (`FDATA`).
+#'
+#' @details
+#' `NM-TRAN` is a preprocessor for `NONMEM` that translates user-specified
+#' control stream data and instructions into a form executable by `NONMEM`.
+#'
+#' `run_nmtran()` allows users to test their models ahead of submission to ensure
+#' correct coding.
+#'
+#' @section Supported `bbi_args`:
+#' `run_nmtran()` supports passing the following raw `NMFE` options through to `NM-TRAN`
+#' via the familiar `.bbi_args` argument.
+#'  - **`maxlim`**: Set the maximum values for the buffers used by `NONMEM`
+#'  (if 0, don't pass `-maxlim` to nmfe) (default is 2)
+#'    - If `maxlim = 3`, it is preferred to also set `tprdefault = TRUE`, but
+#'    _not_ `prdefault`, as `NM-TRAN`'s optional resizing of the `PREDPP` size
+#'    parameter `MAXRECID` may conflict with the `-prdefault` option.
+#'  - **`prdefault`**: If `TRUE`, do not recompile any routines other than `FSUBS`
+#'  - **`tprdefault`**: If `TRUE`, test if is okay to do `-prdefault`
+#'
+#' @param .mod A `bbr` model object.
+#' @param .bbi_args A named list specifying arguments to pass to `NM-TRAN`.
+#'   Similar to the `.bbi_args` argument defined in [submit_model()], though here
+#'   only `prdefault`, `tprdefault`, and `maxlim` flags are passed to `NM-TRAN`.
+#'   See details.
+#' @param .config_path Path to a bbi configuration file. If `NULL`, the default,
+#'   will attempt to use a `bbi.yaml` in the same directory as the model.
+#' @param nmtran_exe Path to an `NM-TRAN` executable. If `NULL`, will look for a
+#'   `bbi.yaml` file in the same directory as the model.
+#' @param delete_on_exit Logical. If `FALSE`, don't delete the temporary folder
+#'   containing the `NM-TRAN` run, which will be stored in the current working
+#'   directory.
+#'
+#' @examples
+#' \dontrun{
+#' mod <- read_model(file.path(MODEL_DIR, 1))
+#' run_nmtran(mod)
+#'
+#' # Set the path to an NM-TRAN executable
+#' run_nmtran(mod, nmtran_exe = "/opt/NONMEM/nm75/tr/NMTRAN.exe")
+#'
+#' }
+#' @name nmtran
+NULL
+
+
+#' @describeIn nmtran Run `NM-TRAN` on a model object to validate its control
```

Comment:
This `describeIn` use is odd given the fdata functionality removal.

Reply:
I agree, but opted to keep this setup since the fdata feature should be added soon after, and this shouldnt affect the user in any notable way

Reply:
I'd prefer you approach this PR without making assumptions about when or even if the fdata PR will land.  That avoids odd bits sticking around if things don't land as you expect.  It doesn't take much time compared to what you'd already do when adjusting for the fdata addition.

But yes, it's minor.  As you wish.

Reply:
Made the adjustment